src/server.ts

```diff
@@ -41,43 +39,32 @@ export interface OperationMessage {
   type: string;
 }
 
-export type ExecuteReactiveFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => IObservable<ExecutionResult>;
-
-export type ExecuteFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => Promise<ExecutionResult>;
-
-export type SubscribeFunction = (
-  schema: GraphQLSchema,
-  document: DocumentNode,
-  rootValue?: any,
-  contextValue?: any,
-  variableValues?: {[key: string]: any},
-  operationName?: string,
-) => AsyncIterator<ExecutionResult>;
-
-export interface Executor {
-  execute?: ExecuteFunction;
-  executeReactive?: ExecuteReactiveFunction;
-  subscribe?: SubscribeFunction;
-}
+export type ExecuteFunction = (schema: GraphQLSchema,
+                               document: DocumentNode,
+                               rootValue?: any,
+                               contextValue?: any,
+                               variableValues?: { [key: string]: any },
+                               operationName?: string) => Promise<ExecutionResult> | AsyncIterator<ExecutionResult>;
```

Comment:
@leebyron 
Yeah, the original `execute` of `graphql-js` returns a `Promise`.
I talked to @DxCx and we think that if we allow `execute` to return `AsyncIterator` in this package, we can also support Hagai's implementation for reactive graphql (https://github.com/apollographql/subscriptions-transport-ws/pull/108#issuecomment-302083499)